ledger/ledger-api-test-tool/src/main/scala/com/daml/ledger/api/testtool/infrastructure/LedgerSession.scala

```diff
@@ -23,7 +23,9 @@ private[testtool] final class LedgerSession private (
 
   private[this] val logger = LoggerFactory.getLogger(classOf[LedgerSession])
 
-  private[this] val bindings: LedgerBindings = new LedgerBindings(channel, config.commandTtlFactor)
+  val services: LedgerServices = new LedgerServices(channel)
+
+  val bindings: LedgerBindings = new LedgerBindings(services, config.commandTtlFactor)
```

Comment:
Having read the following code I see why you made this change. I would still recommend to revert it. Rather, I would suggest to expose them through the `LedgerTestContext` and the `LedgerTestSuite`.

Reply:
This would also allow to expose a method that does not require to take an application identifier and command identifier, making it shorter and less prone to error.

Reply:
The exposed `submitAndWait` can also take a `Template` rather then a command to make the usage even shorter.

Reply:
Would you then expose these methods with the possibility to override the ledger id (i.e. provide one that is very likely to fail) just for this test? I think we'll probably need to go down directly to the services for other tests as well.

Reply:
I would create a method that explicitly allows to create invalid commands, e.g. with an invalid ledger identifier and/or a repeated command identifier.

Reply:
Your point makes a lot of sense, it took me a while today to get to it. I solved this by allowing the client to pass `pollutants`, i.e. lens transformation that can mutate an outgoing request. The end result seems reasonable. PR coming soon.